plugins/autoenv/autoenv.plugin.zsh

```diff
@@ -13,6 +13,7 @@ if ! type autoenv_init >/dev/null; then
       ~/.autoenv
       ~/.local/bin
       /usr/local/opt/autoenv
+      /opt/homebrew/opt/autoenv
```

Comment:
Looks nice, we have done some changes to support M1 brew folder structure, for example in #10618, so this looks as a natural continuation of that.